sql/src/main/java/io/crate/action/sql/TransportSQLAction.java

```diff
@@ -49,141 +42,189 @@
 import org.elasticsearch.transport.TransportService;
 
 import javax.annotation.Nonnull;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 
 
 @Singleton
-public class TransportSQLAction extends TransportBaseSQLAction<SQLRequest, SQLResponse> {
+public class TransportSQLAction extends TransportAction<SQLRequest, SQLResponse> {
 
     private static final DataType[] EMPTY_TYPES = new DataType[0];
     private static final String[] EMPTY_NAMES = new String[0];
-    private static final Object[][] EMPTY_ROWS = new Object[0][0];
+    private static final Object[][] EMPTY_ROWS = new Object[0][];
+
+    private final static String UNNAMED = "";
+    final static int DEFAULT_SOFT_LIMIT = 10_000;
+
+    private final SQLOperations sqlOperations;
 
     @Inject
     public TransportSQLAction(
-            ClusterService clusterService,
-            Settings settings,
-            ThreadPool threadPool,
-            Analyzer analyzer,
-            Planner planner,
-            Provider<Executor> executor,
-            TransportService transportService,
-            StatsTables statsTables,
-            ActionFilters actionFilters,
-            IndexNameExpressionResolver indexNameExpressionResolver,
-            TransportKillJobsNodeAction transportKillJobsNodeAction) {
-        super(clusterService, settings, SQLAction.NAME, threadPool,
-            analyzer, planner, executor, statsTables, actionFilters,
-            indexNameExpressionResolver, transportKillJobsNodeAction,
-            transportService.getTaskManager());
-
+        SQLOperations sqlOperations,
```

Comment:
would prefer moving 1st argument to ctor line and so the indention would result in better readbility